control-plane/plugin/src/resources/volume.rs

```
@@ -192,6 +193,32 @@ impl Scale for Volume {
     }
 }
 
+#[async_trait(?Send)]
+impl Set for Volume {
+    type ID = VolumeId;
+    async fn set(id: &Self::ID, prop_name: &str, prop_value: &str, output: &utils::OutputFormat) {
+        match RestClient::client()
+            .volumes_api()
+            .put_volume_prop(id, prop_name, prop_value)
+            .await
+        {
+            Ok(volume) => match output {
+                OutputFormat::Yaml | OutputFormat::Json => {
+                    // Print json or yaml based on output format.
+                    utils::print_table(output, volume.into_body());
+                }
+                OutputFormat::None => {
+                    // In case the output format is not specified, show a success message.
+                    println!("Volume {id} max_snapshots set to {prop_value} successfully 🚀")
+                }
+            },
+            Err(e) => {
+                println!("Failed to scale volume {id}. Error {e}")
```

should return an error now